ui/app-loader/src/index.ts

```diff
@@ -107,37 +106,68 @@ export default class AppLoader {
   }
 
   private watchEvents() {
-    this.uiEvents.subscribe(async eventData => {
-      if (typeof eventData === 'object' && eventData.event) {
-        switch (eventData.event) {
-          case EventTypes.ExtensionPointMount:
-            this.onExtensionPointMount(eventData.data);
-            break;
-          case EventTypes.ExtensionPointUnmount:
-            this.onExtensionPointUnmount(eventData.data);
-            break;
-          case EventTypes.ModalMount:
-            this.onModalMount(eventData.data);
-            break;
-          case EventTypes.ModalUnmount:
-            this.onModalUnmount(eventData.data);
-            break;
-          case EventTypes.InstallIntegration:
-            await this.installIntegration(eventData.data);
-            break;
-          case EventTypes.UninstallIntegration:
-            await this.uninstallIntegration(eventData.data);
-            break;
-          default:
-            break;
-        }
-      }
-    });
+    const filterEvent = (eventType: EventTypes) => {
+      return filter((eventData: UIEventData) => eventData.event === eventType);
+    };
+
+    this.uiEvents
+      .pipe(filterEvent(EventTypes.ExtensionPointMount))
+      .pipe(map(event => event.data))
+      .subscribe({
+        next: this.onExtensionPointMount.bind(this),
+        error: err =>
+          this.loaderLogger.error(`Error in extension mount subscription: ${err.message}`),
+      });
+
+    this.uiEvents
+      .pipe(filterEvent(EventTypes.ExtensionPointUnmount))
+      .pipe(map(event => event.data))
+      .subscribe({
+        next: this.onExtensionPointUnmount.bind(this),
+        error: err =>
+          this.loaderLogger.error(`Error in extension unmount subscription: ${err.message}`),
+      });
+
+    this.uiEvents
+      .pipe(filterEvent(EventTypes.ModalMount))
+      .pipe(map(event => event.data))
+      .subscribe({
+        next: this.onModalMount.bind(this),
+        error: err => this.loaderLogger.error(`Error in modal mount subscription: ${err.message}`),
+      });
+
+    this.uiEvents
+      .pipe(filterEvent(EventTypes.ModalUnmount))
+      .pipe(map(event => event.data))
+      .subscribe({
+        next: this.onModalUnmount.bind(this),
+        error: err =>
+          this.loaderLogger.error(`Error in modal unmount subscription: ${err.message}`),
+      });
+
+    this.uiEvents
+      .pipe(filterEvent(EventTypes.InstallIntegration))
+      .pipe(map(event => event.data))
+      .subscribe({
+        next: this.installIntegration.bind(this),
+        error: err =>
+          this.loaderLogger.error(`Error in install integration subscription: ${err.message}`),
+      });
+
+    this.uiEvents
```

Comment:
should there also be events for enable/disable integration?

Reply:
it will, but not in this pr

Reply:
it requires more work and changes to mount/unmount logic